src/pyfaf/repos/dnf.py

```diff
@@ -87,34 +94,38 @@ def list_packages(self, architectures):
         release, arch, srpm_name, type, filename, url items.
         """
 
-        self.yum_base.arch.archlist = architectures
-
         result = []
         try:
-            packagelist = self.yum_base.doPackageLists('all', showdups=True)
-        except yum.Errors.RepoError as err:
+            packagelist = self.dnf_base.sack.query().filterm(arch=architectures)
+        except dnf.exceptions.RepoError as err:
             self.log_error("Repository listing failed: '{0}'".format(err))
             return result
 
-        pkgs = (packagelist.available +
-                packagelist.old_available +
-                packagelist.reinstall_available)
+# Get available packages
+###############################################################################
```

Comment:
Thinking about this - Will dnf ever provide `old_available` and `reinstall_available`? If so, please add BZ number here, that requests this feature. But also, if it is not necessary, you may remove this comment.

Reply:
I have opened [BZ 1614666](https://bugzilla.redhat.com/show_bug.cgi?id=1614666).